plugins/AudioFileProcessor/AudioFileProcessorWaveView.cpp

```diff
@@ -376,14 +382,16 @@ void AudioFileProcessorWaveView::zoom(const bool out)
 void AudioFileProcessorWaveView::slide(int px)
 {
 	const double fact = qAbs(double(px) / width());
-	auto step = range() * fact * (px > 0 ? -1 : 1);
+	double step = range() * fact * (px > 0 ? 1 : -1);
+
+	// get the real start and end frame
+	const auto sampleStart = static_cast<double>(m_sample->startFrame());
+	const auto sampleEnd = static_cast<double>(m_sample->endFrame());
```

Comment:
Wanted to point out that comments should really only be used scarcely IMO. For simple code like that, I can quickly understand what it does. Using good names like `sampleStart` and `sampleEnd` also really help with understanding the code already.